src/python/pants/backend/go/util_rules/build_pkg_target_test.py

```diff
@@ -353,3 +390,58 @@ def test_build_invalid_target(rule_runner: RuleRunner) -> None:
     assert dep_build_request.request is None
     assert dep_build_request.exit_code == 1
     assert "dep/f.go:1:1: expected 'package', found invalid\n" in (dep_build_request.stderr or "")
+
+
+def test_build_codegen_target(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "go.mod": dedent(
+                """\
+                module example.com/greeter
+                go 1.17
+                """
+            ),
+            "generate_from_me.txt": "",
+            "greeter.go": dedent(
+                """\
+                package greeter
+
+                import "fmt"
+                import "codegen.com/gen"
+
+                func Hello() {
+                    fmt.Println(gen.Quote("Hello world!"))
+                }
+                """
+            ),
+            "BUILD": dedent(
+                """\
+                go_mod(name='mod')
+                go_package(name='pkg', dependencies=[":gen"])
+                file(name='gen', source='generate_from_me.txt')
+                """
+            ),
+        }
+    )
+
+    # Running directly on a codegen target should work.
+    assert_pkg_target_built(
```

Comment:
Note that this triggers actual Go compilation - it works 🙌

Reply:
Nice!